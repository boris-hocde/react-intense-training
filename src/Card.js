import React from 'react'

const Card = ({ picture, title, text, footer }) =>
  <div className="card">
    <img className="card-img-top" src={picture} alt={title} width="100%" />
    <div className="card-block">
      <h4 className="card-title">
        {title}
      </h4>
      <div className="card-text">
        {text}
      </div>
    </div>
    <div className="card-footer text-muted">
      {footer}
    </div>
  </div>

export default Card
