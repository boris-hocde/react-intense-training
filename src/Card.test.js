import React from 'react'
import renderer from 'react-test-renderer'
import Card from './Card'

describe('Card', () => {
  it('should render a card', () => {
    const component = renderer.create(
      <Card
        picture="/picture"
        title="Card title"
        text="Card text"
        footer="Card footer"
      />,
    )
    let tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
})
