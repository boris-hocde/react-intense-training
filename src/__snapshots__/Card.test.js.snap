// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should render a card 1`] = `
<div
  className="card sc-bdVaJa fxIjnI"
>
  <img
    alt="Card title"
    className="card-img-top"
    src="/picture"
    width="100%"
  />
  <div
    className="card-block"
  >
    <h4
      className="card-title"
    >
      Card title
    </h4>
    <div
      className="card-text"
    >
      Card text
    </div>
  </div>
  <div
    className="card-footer text-muted"
  >
    Card footer
  </div>
</div>
`;
